### PR TITLE
Add weavenet metrics to billing

### PR DESCRIPTION
**Pull request by leth**
This is intended to give some insights into adoption among weave cloud users

### CODE REVIEW

**Comment by leth** [comment]
The build for this is going to fail until the vendor-ed code is updated to contain https://github.com/weaveworks/billing-client/pull/5

File: app/multitenant/billing_emitter.go [hunk 1]
```diff
@@ -112,6 +118,20 @@ func (e *BillingEmitter) reportInterval(r report.Report) time.Duration {
 	return d
 }
 
+// Tries to determine if this report came from a host running Weave Net
+func hasWeaveNet(r report.Report) bool {
+	hasNet := false
+	for _, n := range r.Overlay.Nodes {
+		overlayType, _ := report.ParseOverlayNodeID(n.ID)
+		if overlayType == report.WeaveOverlayPeerPrefix {
```
Comment (2opremio) [inline app/multitenant/billing_emitter.go]: More compact:

```
if hasNet = (overlayType == report.WeaveOverlayPeerPrefix); hasNet {
  break
}
```

Comment (leth) [inline app/multitenant/billing_emitter.go]: Cool! Thanks for the pointer, I'm only just learning go :)

**Comment by 2opremio** [comment]
Please make sure linting/tests pass: https://circleci.com/gh/weaveworks/scope/7149?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by 2opremio** [comment]
> The build for this is going to fail until the vendor-ed code is updated to contain weaveworks/billing-client#5

Gotcha. Please ping me once the vendored code is updated

File: app/multitenant/billing_emitter.go (hunk 1 above)
Comment (2opremio) [inline app/multitenant/billing_emitter.go]: You are doing really well! :)

Comment (rade) [inline app/multitenant/billing_emitter.go]: you might as well get rid of `hasNet` and `return true` instead.

**Comment by leth** [comment]
Yeah can do! I'd heard the go convention was to have one exit point, but
perhaps that's more of a concern in longer functions.

**Comment by tomwilkie** [comment]
>  I'd heard the go convention was to have one exit point

Really?  Where did you read this?  I'd say quite the opposite.

**Comment by leth** [comment]
I think it was something I read ages ago; I may have imagined it. I'll simplify :)

**Comment by leth** [comment]
@2opremio I've updated billing-client

**Pull request merged**